src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,774 @@
+
+        ### Regular Trades
+
+        # Binance requires a symbol/market
+        # max limit is 1000
+        for market in self.markets:
+            since = self.start_time_ms
+            test = True
+            #       while True:
+            while test:
+                market_trades = self.client.fetch_my_trades(symbol=market, since=since, limit=1000)
+                #   {
+                #       'info':         { ... },                    // the original decoded JSON as is
+                #       'id':           '12345-67890:09876/54321',  // string trade id
+                #       'timestamp':    1502962946216,              // Unix timestamp in milliseconds
+                #       'datetime':     '2017-08-17 12:42:48.000',  // ISO8601 datetime with milliseconds
+                #       'symbol':       'ETH/BTC',                  // symbol
+                #       'order':        '12345-67890:09876/54321',  // string order id or undefined/None/null
+                #       'type':         'limit',                    // order type, 'market', 'limit' or undefined/None/null
+                #       'side':         'buy',                      // direction of the trade, 'buy' or 'sell'
+                #       'takerOrMaker': 'taker',                    // string, 'taker' or 'maker'
+                #       'price':        0.06917684,                 // float price in quote currency
+                #       'amount':       1.5,                        // amount of base currency
+                #       'cost':         0.10376526,                 // total cost, `price * amount`,
+                #       'fee':          {                           // provided by exchange or calculated by ccxt
+                #           'cost':  0.0015,                        // float
+                #           'currency': 'ETH',                      // usually base currency for buys, quote currency for sells
+                #           'rate': 0.002,                          // the fee rate (if available)
+                #       },
+                #   }
+
+                # * The work on ``'fee'`` info is still in progress, fee info may be missing partially or entirely, depending on the exchange capabilities.
+                # * The ``fee`` currency may be different from both traded currencies (for example, an ETH/BTC order with fees in USD).
+                # * The ``cost`` of the trade means ``amount * price``. It is the total *quote* volume of the trade (whereas ``amount`` is the *base* volume).
+                # * The cost field itself is there mostly for convenience and can be deduced from other fields.
+                # * The ``cost`` of the trade is a *"gross"* value. That is the value pre-fee, and the fee has to be applied afterwards.
+                for trade in market_trades:
+                    self._process_sell(trade, out_transactions)

```

Comment:
I see you have a debug log inside _process_sell and _process_buy: it would be good to handle all debug logs at the higher level, so that lower-level functions don't have to worry about it, but you don't have to do it, it's just a suggestion. The important thing is to capture all native data in the debug log.